code/mathematical_algorithms/src/exponentiation_power/exponentiation_by_squaring/exponentiation_by_squaring.c

```diff
@@ -0,0 +1,30 @@
+#include <stdio.h>
+
+int 
+power(int num, int exp)
+{
+	if (exp == 0)
+		return (1);
+	if (exp == 1) 
+		return (num);
+
+	int temp = power(num, exp / 2);
+	temp *= temp;
+	if (exp % 2 ==1 )
+		temp *= num;
+
+	return (temp);
+}
+
+int 
+main()
+{
+	int num, exp;
+	printf("Enter Number \n");
+	scanf("%d", &num);
+	printf("Enter Exponent \n");
+	scanf("%d", &exp);
+	printf("Result: %d \n", power(num, exp));
+
+	return (0);
+}
```

Comment:
Missing newline at end of file.

Reply:
Please separate src and test code.

Reply:
What do you mean by separating src and test code? Can you show me once how it is done?

Reply:
Since the code is API for coders, if one include this file, he/she will get an compile error for that redefinition of the main function.

```
#ifdef COSMOS_DEBUG

int 
main()
{
	int num, exp;
	printf("Enter Number \n");
	scanf("%d", &num);
	printf("Enter Exponent \n");
	scanf("%d", &exp);
	printf("Result: %d \n", power(num, exp));

	return (0);
} 

#endif /*COSMOS_DEBUG*/
```

or

create file and name it by test_exponentialiation_by_squaring.c, and put into test folder
```
#include "exponentialiation_by_squaring.c"

int
main()
{
    ...
}
```